src/main/java/seedu/exercise/logic/commands/statistic/StatsCommand.java

```diff
@@ -0,0 +1,63 @@
+package seedu.exercise.logic.commands.statistic;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_CHART;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_ENDDATE;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_STARTDATE;
+
+import seedu.exercise.logic.commands.Command;
+import seedu.exercise.logic.commands.CommandResult;
+import seedu.exercise.model.Model;
+import seedu.exercise.model.ReadOnlyResourceBook;
+import seedu.exercise.model.property.Date;
+import seedu.exercise.model.resource.Exercise;
+
+/**
+ * Generate statistic with given parameters.
+ */
+public class StatsCommand extends Command {
+
+    public static final String COMMAND_WORD = "stats";
+
+    public static final String MESSAGE_USAGE = "Parameters: "
+        + PREFIX_CATEGORY + "CATEGORY "
+        + PREFIX_CHART + "CHART TYPE "
+        + PREFIX_STARTDATE + "START DATE "
+        + PREFIX_ENDDATE + "END DATE " + "\n"
+        + "Example: " + COMMAND_WORD + " "
+        + PREFIX_CATEGORY + "calories "
+        + PREFIX_CHART + "barchart "
+        + PREFIX_STARTDATE + "30/03/2019 "
+        + PREFIX_ENDDATE + "05/04/2019 ";
+
+    private String chart;
+    private String category;
+    private Date startDate;
+    private Date endDate;
+
+    /**
+     * Creates a StatsCommand to generate statistic.
+     */
+    public StatsCommand(String chart, String category, Date startDate, Date endDate) {
+        this.chart = chart;
+        this.category = category;
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        ReadOnlyResourceBook<Exercise> exercises = model.getExerciseBookData();
+        StatsFactory statsFactory = new StatsFactory(exercises, chart, category, startDate, endDate);
+        Statistic statistic = statsFactory.generateStatistic();
+        model.updateStatistic(statistic);
+        return new CommandResult("Chart displayed");
```

Comment:
May want to store the message in CommandResult as a private final String MESSAGE_SUCCESS in this class.

Reply:
done